pkg/api/norman/customization/cluster/utils.go

```diff
@@ -151,3 +153,10 @@ func CanCreateRKETemplate(callerID string, subjectAccessReviewClient clientauthv
 	logrus.Debugf("CanCreateRKETemplate: %v", result)
 	return result.Status.Allowed, nil
 }
+
+func updateClusterWithRetryOnConflict(clusterClient v3.ClusterInterface, cluster *v3.Cluster) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
@dramich [`retry.DefaultRetry`](https://github.com/kubernetes/client-go/blob/fb61a7c88cb9f599363919a34b7c54a605455ffc/util/retry/util.go#L26-L33) currently tries to retry 5 times with 10ms in between. Should this be modified? Right now, this makes us somewhat more resilient but the error still appears if you hammer the API enough. The Terraform provider will also be modified to do retries based on @rawmind0 's comment [here](https://github.com/rancher/rancher/issues/30188#issuecomment-733211942).

It will only ever need to retry if the underlying object is being modified at the same time.

Reply:
The function needs to refresh the cluster, otherwise it's trying with the old version of the object which is why the error still appears.

Reply:
good catch. I updated the function to take in an `updateFunc` that allows the caller to provide a function that performs the changes on the latest version of the cluster.